sktime/transformations/series/temporian.py

```diff
@@ -0,0 +1,123 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Implementation of TemporianTransformer. Contributed by the Temporian team."""
+
+__author__ = ["ianspektor"]  # TODO: add other Temporian devs who contributed to this
+
+from sktime.transformations.base import BaseTransformer
+
+
+# TODO: add usage example to docstring
+# see https://www.sktime.net/en/latest/developer_guide/dependencies.html#dependencies
+# for how to handle soft deps (temporian) there (need to not run doc tests on it)
+class TemporianTransformer(BaseTransformer):
+    """Applies a Temporian function to the input time series.
+
+    This transformer applies a [Temporian](https://temporian.readthedocs.io/en/stable/)
+    function to the input time series.
+
+    The conversion from sktime's internal representation to Temporian's
+    [EventSet](https://temporian.readthedocs.io/en/stable/reference/temporian/EventSet/)
+    and back is handled automatically by the transformer.
+
+    Parameters
+    ----------
+    function : Callable[[temporian.EventSet], temporian.EventSet]
+        Temporian function to apply to the input time series. The function must receive
+        and return a single Temporian EventSet, and can apply an arbitrary number of
+        Temporian operators to its input.
+    """
+
+    _tags = {
+        "univariate-only": False,
+        "authors": [
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who contributed to this
+        "maintainers": [
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who will maintain this
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "scitype:transform-labels": "None",
+        # TODO: extend to pd-multiindex and pd_multiindex_hier
+        "X_inner_mtype": "pd.DataFrame",
+        "fit_is_empty": True,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+        "python_dependencies": ["temporian"],
+        "python_version": "<3.12",
+    }
+
+    def __init__(self, function):
+        self.function = function
+
+        super().__init__()
+
+        # TODO: ensure function receives a single EventSet/param with inspect module?
+        #       or is failing in runtime OK?
+        # TODO: @tp.compile the function
```

Comment:
Is there value to support both @tp.compile and no-@tp.compile functions? In other words, it is the user that decides if the function is compiled. A complex function might be easier to write/debug if not compiled, since the user can print intermediate results. Support for EventSet (!=EventSetNode) computation (e.g., computation that depends on the total number of events) might also be valuable in some cases.

After all, from the caller point of view, if we don't consider serialization and distributed computation, @tp.compile and no-@tp.compile functions have the same semantics.

Reply:
could be a parametmer `compile: bool`?

Reply:
thumbs up to both the rationale and the compile parameter, although not really necessary since we could just tell the user to wrap it themselves in the docstring, I think it'll make the functionality more visible to less advanced users